batch2/batch/driver/scheduler.py

```diff
@@ -38,37 +39,87 @@ def __init__(self, app):
                 await changed.wait()
                 changed.clear()
 
-    async def cancel_1(self):
+    async def get_n_jobs_per_user(self, max_n_jobs):
         records = self.db.execute_and_fetchall(
             '''
+SELECT user, ready_n_jobs, ready_cores_mcpu
+FROM users;
+''')
+
+        ready_users = {record['user']: record['ready_n_jobs'] for record in records}
```

Comment:
I think we should schedule just based on ready cores, not number of ready jobs.  If there are 1000 cores available, and I submit 8 core jobs, and you submit 1 core jobs, we should both get 500 cores.  I think the algorithm should look something like: for each user with with runnable jobs, in ascending order by runnable cores, give the user up to min(needed cores, 1/(remaining users) of the available cores).  You can skip users with no runnable cores (although the algorithm gives the same answer either way).